src/style.toit

```diff
@@ -205,6 +205,243 @@ class BorderWidth:
+    canvas.transform.xywh 0 0 w h: | x2 y2 w2 h2 |
+      right := x2 + w2
+      bottom := y2 + h2
+      if x2           <= 0 and y2 + radius_ <= 0 and right           >= canvas.width_ and bottom - radius_ >= canvas.height_ or
+         x2 + radius_ <= 0 and y2           <= 0 and right - radius_ >= canvas.width_ and bottom           >= canvas.height_:
+        // In the middle, the window content is 100% opaque and draw on top of the
+        // frame.  There is no need to provide a frame alpha map, so for efficiency we
+        // just return 0 which indicates the frame is 100% transparent.
+        return element.ClippingDiv.ALL_TRANSPARENT
+
+    // Transform outer dimensions including border to determine if the canvas
+    // is wholly outside the window and its shadow.
+    extent_helper_: | left top right bottom |
+      canvas.transform.xywh -left -top (w + left + right) (h + top + bottom): | x2 y2 w2 h2 |
+        if x2 + w2 <= 0 or y2 + h2 <= 0 or x2 >= canvas.width_ or y2 >= canvas.height_:
+          return element.ClippingDiv.ALL_TRANSPARENT  // The frame is not opaque outside the shadow
+
+    // Create a bitmap to describe the frame's extent.  It needs to be padded
+    // relative to the canvas size so we can use the Gaussian blur.
+    transparency_map := canvas.make_alpha_map --padding=(blur_radius * 2)
+    transparency_map.transform = (canvas.transform.invert.translate -blur_radius -blur_radius).invert
+
+    max_shadow_opacity := (shadow_opacity_percent * 2.5500001).to_int
+    draw_rounded_corners_ transparency_map drop_distance_x_ drop_distance_y_ w h max_shadow_opacity
+
+    if blur_radius == 0 or transparency_map is not one_byte.OneByteCanvas_:
+      return transparency_map
+
+    one_byte_map := transparency_map as one_byte.OneByteCanvas_
+
+    // Blur the shadow.
+    bitmap.bytemap_blur one_byte_map.pixels_ transparency_map.width_ blur_radius
+
+    // Crop off the extra that was added to blur.
+    transparency_map_unpadded := canvas.make_alpha_map
+    bitmap.blit
+        one_byte_map.pixels_[blur_radius + blur_radius * one_byte_map.width_..]   // Source.
+        (transparency_map_unpadded as one_byte.OneByteCanvas_).pixels_  // Destination.
+        transparency_map_unpadded.width_   // Bytes per line.
+        --source_line_stride=transparency_map.width_
+    return transparency_map_unpadded
+
+  draw_frame canvas/Canvas:
+    canvas.set_all_pixels 0
+
+class RoundedCornerOpacity_:
+  byte_opacity/ByteArray
+  bit_opacity/ByteArray
+  radius/int
+  bitmap_width/int
+  static cache_ := Map.weak
+
+  static get corner_radius/int -> RoundedCornerOpacity_:
+    cached := cache_.get corner_radius
+    if cached: return cached
+    new := RoundedCornerOpacity_.private_ corner_radius
+    cache_[corner_radius] = new
+    return new
+
+  static TABLE_SIZE_ ::= 256
+  // The heights of a top-right quarter circle of radius [TABLE_SIZE_].
+  static QUARTER_CIRCLE_ ::= create_quarter_circle_array_ TABLE_SIZE_
```

Comment:
Type the static.
If it's not a constant, then the type-checker doesn't yet infer the type.